sycl/doc/design/DeviceConfigFile.md

```diff
@@ -0,0 +1,297 @@
+    string aotToolchain = toolchain;
+    string oclocDevice = device;
+}
+
+def : TargetInfo<"TargetA", [AspectCpu, AspectAtomic64], 
+                                  0, [8, 16], "ocloc", "tgtA">; 
+def : TargetInfo<"TargetB", [AspectGpu, AspectFp16],
+                                  0, [8, 16], "ocloc", "tgtB">;
+def : TargetInfo<"TargetC", [AspectEmulated, AspectImage],
+                                   0, [8, 32], "ocloc", "tgtC">;
+```
+Note: backends tested don't allow lists within `TargetInfo` class. I
+_think_ this is a backend limitation, rather than a TableGen limitation.
+Thus, we should be able to lift this limitation in our own backend.
+
+The generated `.inc` file should look like the example below: 
+```c++
+enum AspectEnum {
+  AspectAccelerator = 3,
+  AspectAtomic64 = 28,
+  AspectCpu = 1,
+  AspectCustom = 4,
+  AspectEmulated = 40,
+  AspectExt_intel_device_id = 37,
+  AspectExt_intel_device_info_uuid = 29,
+  AspectExt_intel_free_memory = 36,
+  AspectExt_intel_gpu_eu_count = 19,
+  AspectExt_intel_gpu_eu_count_per_subslice = 23,
+  AspectExt_intel_gpu_eu_simd_width = 20,
+  AspectExt_intel_gpu_hw_threads_per_eu = 33,
+  AspectExt_intel_gpu_slices = 21,
+  AspectExt_intel_gpu_subslices_per_slice = 22,
+  AspectExt_intel_max_mem_bandwidth = 24,
+  AspectExt_intel_mem_channel = 25,
+  AspectExt_intel_memory_bus_width = 39,
+  AspectExt_intel_memory_clock_rate = 38,
+  AspectExt_intel_pci_address = 18,
+  AspectExt_oneapi_bfloat16_math_functions = 35,
+  AspectExt_oneapi_cuda_async_barrier = 34,
+  AspectExt_oneapi_native_assert = 31,
+  AspectExt_oneapi_srgb = 30,
+  AspectFp16 = 5,
+  AspectFp64 = 6,
+  AspectGpu = 2,
+  AspectHost_debuggable = 32,
+  AspectImage = 9,
+  AspectOnline_compiler = 10,
+  AspectOnline_linker = 11,
+  AspectQueue_profiling = 12,
+  AspectUsm_atomic_host_allocations = 26,
+  AspectUsm_atomic_shared_allocations = 27,
+  AspectUsm_device_allocations = 13,
+  AspectUsm_host_allocations = 14,
+  AspectUsm_shared_allocations = 15,
+  AspectUsm_system_allocations = 17,
+};
+    
+struct TargetInfo {
+  bool maySupportOtherAspects;
+  std::vector<unsigned> aspects;
+  std::vector<unsigned> subGroupSizes;
+  std::string aotToolchain;
+  std::string oclocDevice;
+};
+
+std::map<std::string, TargetInfo> targets = {
+    {"TargetA",
+     {{AspectCpu, AspectAtomic64}, 0, {8, 16}, "ocloc", "tgtA"}},
+    {"TargetB",
+     {{AspectGpu, AspectFp16}, 0, {8, 16}, "ocloc", "tgtB"}},
+    {"TargetC",
+     {{AspectEmulated, AspectImage}, 0, {8, 32}, "ocloc", "tgtC"}}};
+```
+
+We also need a header file that includes the `.inc` file generated by the
+TableGen backend. Other backends don't generate the definition of `struct
+TargetInfo`. This might be a limitation of the backends, or a limitation of
+TableGen. In case it cannot be autogenerated by the backend, a possible
+workaround is simply to define the struct in this header file. This header
+file should look like the code below:
```

Comment:
I think it's better to define it ourselves manually in a separate header file because:
- Developers don't need to run any tool to generate it to be able to find the structure.
- Backend implementation is simpler.
I updated the document accordingly.

Reply:
I think it's better to define it ourselves manually in a separate header file because:
- Developers don't need to run any tool to generate it to be able to find the structure.
- Backend implementation is simpler.

I updated the document accordingly.